xapian-applications/omega/configure.ac

```diff
@@ -247,14 +247,15 @@ dnl fnmatch() is used to match ignore pattern.
 AC_CHECK_FUNCS([fnmatch])
 
 dnl check if poppler is available.
-am_save_LIBS="$LIBS"
+save_LIBS="$LIBS"
 AC_LIB_APPENDTOVAR([LIBS], [-lpoppler-cpp])
```

Comment:
Sorry, I missed some issues here before.

Portable build systems seems to be inherently complex, and there's quite a lot of new stuff to learn here too.

Despite `AC_LIB_APPENDTOVAR` being in the `AC_` namespace, this isn't actually a standard autoconf macro, but one that `iconv.m4` pulls in.

I think I'd rather avoid introducing uses of it outside of `iconv.m4` anyway, but in particular it doesn't use `pkg-config` to find the appropriate flags to use, which we really should be doing when that's the preferred way to find the flags for a library, as it is for libpoppler, e.g.:

```
$ pkg-config poppler-cpp --libs-only-l
-lpoppler-cpp
```

That way things should just work with a non-system install of poppler, or on a platform where poppler needs special compiler flags to be used.

There are some autoconf macros for `pkg-config` which you'd use approximately like so:

```
PKG_CHECK_MODULES([POPPLER_CPP], [poppler-cpp], [
    save_LIBS=$LIBS
    LIBS="$LIBS $POPPLER_LIBS"
    ... do the AC_TRY_LINK here so we catch the libc++/libstdc++ incompatibility situation and auto-disable for that ...
    LIBS=$save_LIBS
    ], [lib_poppler=false])
```

But there's more, and I'm starting to wonder if it would work better to find a time to (virtually) sit down together and work through this interactively.

Reply:
This is solved. We discuss it through IRC channel.